### DISCUSSION TITLE
How to fix the huge gas charge with any manipulations?

### DISCUSSION THREAD

**Discussion by DevCores** [hashlips-lab/nft-erc721-collection#394] [Q&A] [answered]
Can you tell me why there is such a huge gas fee? The whole contract has very high gas for all functions?
A transaction for the transfer of NFT costs $ 400 and other collections have about $ 1
What could be the reason for this problem?
![qskbN61h](https://user-images.githubusercontent.com/10381540/186589509-e21eeda5-9652-475b-8e84-8d8e39a0e817.jpg)
![_cXFNLpY](https://user-images.githubusercontent.com/10381540/186589526-950c8923-e3a1-4117-9612-79ebd422602e.jpg)
![ZFy24ubo](https://user-images.githubusercontent.com/10381540/186589572-0a55573f-cb6d-4bc1-973a-6413876792a3.jpg)

**Comment by liarco** [comment, marked as answer]
Hi @DevCores, issues must have all the requested info and should be used for bug reports only. [Discussions](https://github.com/hashlips-lab/nft-erc721-collection/discussions) are bests suited for questions like this.

Did you mint your tokens in big batches? (Like minting 100 tokens at once or more)
If that's the case, then `ERC721A` is known to have pretty high gas fees in that particular scenario: https://chiru-labs.github.io/ERC721A/#/tips?id=batch-size

**Reply by DevCores** [reply]
> Hi @DevCores, issues must have all the requested info and should be used for bug reports only. [Discussions](https://github.com/hashlips-lab/nft-erc721-collection/discussions) are bests suited for questions like this.
> 
> Did you mint your tokens in big batches? (Like minting 100 tokens at once or more) If that's the case, then `ERC721A` is known to have pretty high gas fees in that particular scenario: https://chiru-labs.github.io/ERC721A/#/tips?id=batch-size

Yes, there was quite a large coinage in a row of over 100 pieces. Is there any way to fix this misunderstanding?

**Reply by liarco** [reply]
If you don't have any public function exposing [`_initializeOwnershipAt()`](https://chiru-labs.github.io/ERC721A/#/erc721a?id=_initializeownershipat), then you can only use the main transfer function in order to break those batches into smaller ones.

For instance if you have a batch from 1 to 100, transferring token 50 would cost much less than 99 (the less the distance from the batch start, the lower the transfer cost), and all the following transfers will be lower.
But I would recommend testing this because you might have to do a lot of transfers so make sure you are happy with the ending result before doing it on the mainnet.

**Reply by DevCores** [reply]
> 

The problem is that the contract is already in the main network.

**Reply by DevCores** [reply]
If you transfer every 5 tokens in the chain, will it break it into batches?

**Reply by DevCores** [reply]
We have transferred every 50 tokens and everything is coming back to normal!

**Reply by liarco** [reply]
Sorry for the late reply, I knew the contract was already live on the mainnet, my suggestion was to recreate the same situation on a testnet before transferring all the real tokens and spend money for it. But I'm really happy to hear it solved your problem.
